CMakeLists.txt

```diff
@@ -141,7 +141,12 @@ if(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
         $<$<COMPILE_LANGUAGE:CXX>:-Wextra>
         $<$<COMPILE_LANGUAGE:CXX>:-pipe>)
 elseif(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
-    message(FATAL_ERROR "Clang is currently not supported.")
+    target_compile_options(${PROJECT_NAME} PUBLIC
```

Comment:
I think this needs a version check for minimum clang-16, coroutines are not building on any version lower than that from my prototyping with getting clang to work.

Reply:
Version check added